Server/BackEndMonitoringServer/src/CHardwareStatusSpecification.cpp

```diff
@@ -0,0 +1,29 @@
+#include "stdafx.h"
+
+#include "EMemoryConvertType.h"
+#include "CHardwareStatusSpecification.h"
+
+CHardwareStatusSpecification::CHardwareStatusSpecification(
+	std::chrono::duration<int>
+	pause_duration, const std::string& path_to_file,
+	EMemoryConvertType count_type) :
+	m_pause_duration(pause_duration), m_path_to_file(path_to_file),
+	m_count_type(count_type)
+{ };
+
+CHardwareStatusSpecification::CHardwareStatusSpecification(
+	const CHardwareStatusSpecification& orig) :
+	m_pause_duration(orig.m_pause_duration),
+	m_path_to_file(orig.m_path_to_file),
+	m_count_type(orig.m_count_type)
+{ };
+
+std::chrono::duration<int> 
+CHardwareStatusSpecification::GetPauseDuration() const
+{ return m_pause_duration; }
+
+EMemoryConvertType CHardwareStatusSpecification::GetCountType() const
+{ return m_count_type; }
+
+std::string* CHardwareStatusSpecification::GetPathToSaveFile()
```

Comment:
Should return const because we are returning the field, also other objects could change this field without informing CHardwareStatusSpecification object

Reply:
Done: const string* .... () const;